src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,143 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (isEmpty()) {
+            Node<T> newNode = new Node<>(null, value, null);
+            head = tail = newNode;
+            head.next = tail;
+            tail.prev = head;
```

Comment:
There's no need to set `head.next = tail;` and `tail.prev = head;` since you're already creating a new node with `null` references to the next and previous nodes.